src/Oauth1.php

```diff
@@ -62,29 +59,40 @@ class Oauth1 implements SubscriberInterface
      */
     public function __construct($config)
     {
-        $this->config = Collection::fromConfig($config, [
+        $this->config = [
             'version'          => '1.0',
             'request_method'   => self::REQUEST_METHOD_HEADER,
             'consumer_key'     => 'anonymous',
             'consumer_secret'  => 'anonymous',
             'signature_method' => self::SIGNATURE_METHOD_HMAC,
-        ], ['signature_method', 'version', 'consumer_key', 'consumer_secret']);
-    }
+        ];
 
-    public function getEvents()
-    {
-        return ['before' => ['onBefore', RequestEvents::SIGN_REQUEST]];
+        foreach ($config as $key => $value)
+        {
+            $this->config[$key] = $value;
+        }
     }
 
-    public function onBefore(BeforeEvent $event)
+    /**
+     * Called when the middleware is handled.
+     *
+     * @param callable $handler
+     *
+     * @return \Closure
+     */
+    public function __invoke(callable $handler)
     {
-        $request = $event->getRequest();
+        return function ($request, array $options) use ($handler) {
 
-        // Only sign requests using "auth"="oauth"
-        if ($request->getConfig()['auth'] != 'oauth') {
-            return;
-        }
+            if (isset($options['auth']) && $options['auth'] == 'oauth')
+                $request = $this->onBefore($request);
+
+            return $handler($request, $options);
+        };
+    }
 
+    public function onBefore(RequestInterface $request)
```

Comment:
Should we rename this sand make it private? I don't think this needs to be public anymore with the middleware approach.